confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,701 @@
+
+        Returns:
+            int: Schema ID if known; else None
+
+        """
+        return self.schema_index.get(schema, None)
+
+    def is_registered(self, subject, schema):
+        return schema in self.subject_schemas
+
+
+class SchemaRegistryClient(object):
+    """
+    Schema Registry Client.
+
+    SchemaRegistryClient configuration properties:
+    +--------------------------+----------------------------------------------+
+    | Property name            | Description                                  |
+    +==========================+==============================================+
+    | url                      | Schema Registry url                          |
+    +--------------------------+----------------------------------------------+
+    |                          | Path to CA certificate file used             |
+    | ssl.ca.location          | to verify the Schema Registry's              |
+    |                          | private key,                                 |
+    +--------------------------+----------------------------------------------+
+    |                          | Path to client's private key                 |
+    |                          | (PEM) used for authentication.               |
+    | ssl.key.location         |                                              |
+    |                          | *ssl.certificate.location must also be set.  |
+    +--------------------------+----------------------------------------------+
+    |                          | Path to client's public key (PEM) used for   |
+    |                          | authentication.                              |
+    | ssl.certificate.location |                                              |
+    |                          | *May be set without ssl.key.location if the  |
+    |                          | private key is stored within the PEM as well |
+    +--------------------------+----------------------------------------------+
+    |                          | Client HTTP credentials in the form of       |
+    |                          | ``username:password ``.                      |
+    | basic.auth.user.info     |                                              |
+    |                          |  By default userinfo is extracted from       |
+    |                          |  the URL if present.                         |
+    +--------------------------+----------------------------------------------+
+
+    Args:
+        conf (dict): Schema Registry client configuration
+
+    See Also:
+        http://confluent.io/docs/current/schema-registry/docs/intro.html
+
+    """
+
+    def __init__(self, conf):
+        self._rest_client = _RestClient(conf)
+        self.cache = _SchemaCache()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        if self._rest_client is not None:
+            self._rest_client.close()
+
+    @property
+    def url(self):
```

Comment:
i don't think there's any point exposing this. especially since if failover is implemented, it will be badly named (should be named urls).